cfme/tests/cloud/test_keypairs.py

```diff
@@ -79,9 +72,6 @@ def test_keypair_crud_with_key(openstack_provider):
         openstack_provider.mgmt.api.keypairs.delete(keypair.name)
         pytest.fail('Timed out deleting keypair')
 
-    with pytest.raises(KeyPairNotFound):
```

Comment:
Again this test assertion needs to be replaced somehow, ideally by keeping the exception on navigation failure.